mewarpx/mewarpx/diags_store/particle_diagnostic.py

```diff
@@ -0,0 +1,154 @@
+
+            plot_data_list (list str): list of data to be plotted for each
+            diagnostic step ("particle_position_x", "particle_position_y",
+            "particle_position_z", "particle_momentum_x", "particle_momentum_y",
+            "particle_momentum_z")
+
+            plot_species (list str): list of species names to be plotted,
+            defaults to all species if not specified. Name variable in
+            mepicmi.Species must be set for each species in the simulation
+
+        """
+        self.name = name
+        self.species = species
+        self.data_list = data_list
+        self.write_dir = write_dir
+
+        self.post_process = post_process
+        self.plot_data_list = plot_data_list
+        self.plot_species = plot_species
+
+        if self.name is None:
+            self.name= "diag"
+        if self.species is None:
+            self.species = mwxrun.simulation.species
+        if self.data_list is None:
+            self.data_list = ["position", "momentum", "weighting"]
+        if self.plot_species is None:
+            self.plot_species = []
+            for specimen in self.species:
+                self.plot_species.append(specimen.name)
+
+        super(ParticleDiagnostic, self).__init__(diag_steps=diag_steps, **kwargs)
+        self.add_particle_diag()
+        if self.post_processing:
+            callbacks.installafterstep(self.check_for_end_of_sim)
+
+        if self.post_process:
+            if self.plot_data_list is None:
+                raise UserWarning("Warning: post_process is True,"
+                                "but plot_data_list is None!")
+            else:
+                self.add_post_processing_check_callback()
+
+
+    def add_particle_diag(self):
+        particle_diag = picmi.ParticleDiagnostic(
+            name=self.name,
+            period=self.diag_steps,
+            species=self.species,
+            data_list=self.data_list,
+            write_dir=self.write_dir
+        )
+
+        mwxrun.simulation.add_diagnostic(particle_diag)
+
+    def diag_data_path(self):
+        read_location = None
+        if self.write_dir == '.':
+            read_location = os.getcwd()
+            read_location = os.path.join(read_location, "diags")
+        else:
+            read_location = self.write_dir
+
+        return read_location
+
+    def check_for_end_of_sim(self):
+        if mwxrun.get_it() == mwxrun.simulation.max_steps:
+            self.do_post_processing()
+
+    def do_post_processing(self):
+        if mwxrun.me == 0:
+            read_path = self.diag_data_path()
+            data_dirs = glob.glob(read_path+'/'+self.name+'*')
+
+            if len(data_dirs) == 0:
+                raise RuntimeError(f'No diagnostic data found in {read_path}')
+
+            for datafolder in data_dirs:
+                print('Reading', datafolder, '\n')
+                head, tail = os.path.split(datafolder)
+                step = tail[tail.rfind(self.name[-1])+1:]
```

Comment:
As elegant as this is, I think it breaks if the last character in `name` is a number rather than letter, right?

Usually we proceed steps with an underscore... if that's the case `datafolder.split("_")[-1]` can work...